app/api/events.py

```diff
@@ -115,78 +122,205 @@ def update_event(request: Request, id:str, eventUpdate: EventUpdate, AccessToken
+        "$elemMatch": {"id": token.user_id}}})
 
-    try : 
-        user['participants'][0]
+    if not participant:
+        raise HTTPException(400, "User not joined event!")
+    try:
+        participant['participants']
     except KeyError:
         raise HTTPException(400, "User not joined event!")
 
-    db.events.update_one({'eid' : event['eid']}, { "$pull": { "participants": member } })
+    db.events.update_one({'eid': event['eid']}, {
+        "$pull": {"participants": {"id": token.user_id}}})
+
+    if event['maxParticipants']:
+        if len(event['participants']) >= event['maxParticipants']:
+            return {'max': True}
+
+    return {'max': False}
 
-    return {'max': event['maxParticipants'] }
 
 @router.get('/{id}/joined', dependencies=[Depends(validate_uuid)])
-def is_joined_event(request: Request, id:str, token: AccessTokenPayload = Depends(authorize)):
+def is_joined_event(request: Request, id: str, token: AccessTokenPayload = Depends(authorize)):
     db = get_database(request)
-
     event = get_event_or_404(db, id)
-    # uses event["eid"] instead of casting eid -> UUID 
-    user = db.events.find_one({"eid" : event["eid"]}, {"participants" : {"$elemMatch" : {"id": token.user_id}}})
+
+    # uses event["eid"] instead of casting eid -> UUID
+    user = db.events.find_one({"eid": event["eid"]}, {"participants": {
+        "$elemMatch": {"id": token.user_id}}})
 
     if not user:
-        return {'joined' : False}
+        return {'joined': False}
 
-    try : 
+    try:
         user['participants'][0]
-        return {'joined' : True}
+        return {'joined': True}
+    except KeyError:
+        return {'joined': False}
+
+
+@router.delete('/{id}/removeParticipant/{uid}', dependencies=[Depends(validate_uuid)])
+def remove_participant(request: Request, id: str, uid: str, token: AccessTokenPayload = Depends(authorize_admin)):
+    db = get_database(request)
+    event = get_event_or_404(db, id)
+    member = db.members.find_one({'id': UUID(uid).hex})
+
+    if not member:
+        raise HTTPException(400, "User could not be found")
+
+    participant = db.events.find_one({"eid": event["eid"]}, {"participants": {
+        "$elemMatch": {"id": UUID(uid).hex}}})
+
+    if not participant:
+        raise HTTPException(400, "User not joined event!")
+
+    try:
+        participant['participants']
     except KeyError:
-        return {'joined' : False}
+        raise HTTPException(400, "User not joined event!")
+
+    db.events.update_one({'eid': event['eid']}, {
+                         "$pull": {"participants": {"id": UUID(uid).hex}}})
+
+    return Response(status_code=200)
+
+
+@router.get('/{id}/export', dependencies=[Depends(validate_uuid)])
+async def exportEvent(background_tasks: BackgroundTasks, request: Request, id: str, token: AccessTokenPayload = Depends(authorize_admin)):
+    filename = f"{id}.xlsx"
+    path = get_export_path(request)
+    path = path + filename
```

Comment:
Does the endpoint handle if directory doesn't exist? if so it should handle it :file_folder:

Reply:
Added export path check and mkdir on not found